weasyl/controllers/user.py

```diff
@@ -116,7 +116,7 @@ def signin_2fa_auth_get_(request):
 
     # Only render page if the session exists //and// the password has
     # been authenticated (we have a UserID stored in the session)
-    if sess and '2fa_pwd_auth_userid' not in sess.additional_data:
+    if not sess or not sess.additional_data or '2fa_pwd_auth_userid' not in sess.additional_data:
```

Comment:
It looks like `sess` can never be `None` – does this just need to be `not sess.additional_data`?